src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/ActivitiesMigrationService.kt

```diff
@@ -93,17 +93,19 @@ class ActivitiesMigrationService(
         }
   }
 
-  suspend fun endMigratedActivities(migrationId: String): LocalDate {
+  suspend fun endMigratedActivities(migrationId: String) {
     val activityCount = activitiesMappingService.getMigrationCount(migrationId)
     if (activityCount == 0L) throw NotFoundException("No migrations found for $migrationId")
 
     val allActivityIds = activitiesMappingService.getActivityMigrationDetails(migrationId, activityCount).content
       .map { it.nomisCourseActivityId }
 
     val migration = migrationHistoryService.get(migrationId)
-    val nomisActivityEndDate = objectMapper.readValue(migration.filter, ActivitiesMigrationFilter::class.java).activityStartDate?.minusDays(1) ?: LocalDate.now()
-    nomisApiService.endActivities(allActivityIds, nomisActivityEndDate)
-    return nomisActivityEndDate
+    val filter = objectMapper.readValue(migration.filter, ActivitiesMigrationFilter::class.java)
+    // There will always be a start date because it's now mandatory in the UI, but is still nullable due to old data that can be displayed
+    filter.nomisActivityEndDate = filter.activityStartDate!!.minusDays(1)
+    nomisApiService.endActivities(allActivityIds, filter.nomisActivityEndDate!!)
+    migrationHistoryService.updateFilter(migrationId, filter)
```

Comment:
This is the main change - when ending activities update the filter with the end date